### PR TITLE
Save/Load as handled events

### PR DESCRIPTION
**Pull request by Wyqer**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| Needs wipe? | yes |
| Fixed issues | #502 |

### Description:
Saving and loading is now based on events and not a function which is calling module specific functions.
This way it's more easy to have a module as standalone without the need to edit existing files.
The event handlers for load/save needs to be registered in a preInitModule function, as it needs to be already available upon the postInit load and save.

### Content:
- [x] Save/Load as events
- [x] Init module adjusted accordingly

### Tested on:
- [x] Local MP Vanilla
- [x] ~~Local MP ACE~~
- [x] Dedicated MP Vanilla
- [x] ~~Dedicated MP ACE~~

### CODE REVIEW

File: Missionframework/modules/00_init/fnc/fn_init_loadData.sqf [hunk 1]
```diff
@@ -17,12 +17,18 @@
...
 
-params [["_moduleData",[]]];
-
 if (KPLIB_param_debugSave > 0) then {diag_log "[KP LIBERATION] [SAVE] Init module loading...";};
 
+private _moduleData = [];
+
+if (!isNil "KPLIB_save_data") then {
+    _moduleData = KPLIB_save_data select {(_x select 0) == "init"};
+};
```
Comment (veteran29) [inline Missionframework/modules/00_init/fnc/fn_init_loadData.sqf]: As this is supposed to be also used by external modules the whole public API should be done via functions.

Something like `["init"] call KPLIB_fnc_init_getSaveData;` should be enough. The function should return module data or empty array if no module data was present in the save.

File: Missionframework/modules/00_init/fnc/fn_init_saveData.sqf [hunk 2]
```diff
@@ -14,16 +14,19 @@
...
-    KPLIB_sectors_blufor, // Blufor sectors
-    KPLIB_sectors_fobs apply {getMarkerPos _x} // FOB positions
+// Push module data to the save data array with module name prefix
+KPLIB_save_data pushBack ["init",
+    [
+        date, // Current date and time
+        KPLIB_sectors_lockedVeh, // Locked Vehicles array
+        KPLIB_sectors_blufor, // Blufor sectors
+        KPLIB_sectors_fobs apply {getMarkerPos _x} // FOB positions
+    ]
```
Comment (veteran29) [inline Missionframework/modules/00_init/fnc/fn_init_saveData.sqf]: Same thing as with loading, public API should be done via method so input can be validated, for eg. `[<STRING>, <ARRAY>] call KPLIB_fnc_init_setSaveData;`.

File: Missionframework/modules/00_init/fnc/fn_init_loadData.sqf (hunk 1 above)
Comment (Wyqer) [reply Missionframework/modules/00_init/fnc/fn_init_loadData.sqf]: Sounds sensible, of course 👍

File: Missionframework/modules/00_init/fnc/fn_init_saveData.sqf (hunk 2 above)
Comment (Wyqer) [reply Missionframework/modules/00_init/fnc/fn_init_saveData.sqf]: Same as above ^^

**Pull request merged**